### PR TITLE
Add e2e test for metric extraction

### PR DESCRIPTION
**Pull request by palourde**
Signed-off-by: Simon Plourde <simon@sensu.io>

## What is this change?

It adds an e2e test for metric extraction functionality.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/1389

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

Non!

## Were there any complications while making this change?

Non!

### CODE REVIEW

**Comment by nikkiattea** [comment]
😢 
```
--- FAIL: TestMetricExtraction (18.33s)
	assertions.go:239: 
                          
	Error Trace:	metric_extraction_test.go:53
		
	Error:      	Should not be zero, but was 0
panic: runtime error: index out of range [recovered]
panic: runtime error: index out of range
```

File: testing/e2e/metric_extraction_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		"--command", "echo 'PING ok - Packet loss = 0% | percent_packet_loss=0'",
+		"--metric-format", "nagios_perfdata",
+	)
+	require.NoError(t, err, string(out))
+
+	// FIXME: Give it few seconds to make sure we're not publishing check requests.
+	time.Sleep(15 * time.Second)
+
+	// There should be a stored event for our metric
+	out, err = sensuctl.run("event", "info", agent.ID, "nagios-metric")
```
Comment (nikkiattea) [inline testing/e2e/metric_extraction_test.go]: I think your agent id is `TestMetricExtraction` not `nagios-metric`, hence no metric points showing up in the failed test.

File: testing/e2e/metric_extraction_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	agent, cleanup := newAgent(agentConfig, sensuctl, t)
+	defer cleanup()
+
+	// Create a check that publish check requests
+	out, err := sensuctl.run(
+		"check", "create", "nagios-metric",
+		"--publish",
+		"--interval", "1",
+		"--subscriptions", "test",
+		"--command", "echo 'PING ok - Packet loss = 0% | percent_packet_loss=0'",
```
Comment (nikkiattea) [inline testing/e2e/metric_extraction_test.go]: This is the failure on windows only: `echo`

Comment (nikkiattea) [reply testing/e2e/metric_extraction_test.go]: We could create a go program in tools https://github.com/sensu/sensu-go/tree/master/tools that will allow us to write a string to stdout. That way, the check will be executed properly on both platforms

Comment (palourde) [reply testing/e2e/metric_extraction_test.go]: Ah, good catch!
